timemachine/cpp/src/kernels/k_barostat.cuh

```diff
@@ -41,9 +41,20 @@ void __global__ k_rescale_positions(
         RealType displacement_y = ((centroid_y - center_y) * scale) + center_y - centroid_y;
         RealType displacement_z = ((centroid_z - center_z) * scale) + center_z - centroid_z;
 
-        coords[atom_idx * 3 + 0] += displacement_x;
-        coords[atom_idx * 3 + 1] += displacement_y;
-        coords[atom_idx * 3 + 2] += displacement_z;
+        // centroid of the new molecule
+        centroid_x += displacement_x;
+        centroid_y += displacement_y;
+        centroid_z += displacement_z;
+
+        // compute displacement needed to shift centroid back into the scaled homebox
```

Comment:
Actually I just realized it's not safe to use scaled box here - since its only written by the 0th thread.

Reply:
(so instead we compute it from box * scale)